### ISSUE TITLE
Issue with float value multiplied by 10,000

### ISSUE THREAD

**Issue by himanshu-kumar-lilly** [python/cpython#122391]
# Bug report

### Bug description:

```python
col = "170.92"

val = float(col)*10000

print(val)
```

val = 1709199.9999999998

The output value is incorrect, expected value should be 17092000.0. It's happening only for the above-mentioned scenario.

### CPython versions tested on:

3.8

### Operating systems tested on:

Windows

**Comment by hugovk** [comment]
This is how floating-point arthmetric works, see https://docs.python.org/3/faq/design.html#why-are-floating-point-calculations-so-inaccurate

It's not unique to Python, for example, here's JavaScript:

```js
> col = "170.92"
'170.92'
> val = parseFloat(col)*10000
1709199.9999999998
```

**Comment by rhettinger** [comment]
> The output value is incorrect, expected value should be 17092000.0. 

Sometimes our expectations don't match the realities ;-)   In particular, the above expectation is based on a mental model for floats being a **decimal fraction** where `170.92` is equivalent to `17092 / 10 ** 2`.  And in n fact, we have a module that does what you expect:

```
>>> from decimal import Decimal
>>> col = "170.92"
>>> value = Decimal(col) * 10000
>>> print(value)
1709200.00
```

However, the default *float* type in Python is a **binary fraction** implemented as C double where `170.92` is equivalent to `6013712877423165 / 2 ** 45` which is almost but not exactly equal to `170.92`.   While this doesn't match your expectations, it does have the compelling advantage of hardware support.  That makes it very fast and space-efficient.   Most (but not all) languages and calculators choose binary fractions for those reasons.

In the end, you either need to adjust your expectations (adapt your mental model to what the computer is actually doing) or adjust your tooling (using the decimal module to make the computer adapt to your mental model).
